src/trans/gpu/external/setup_trans.F90

```diff
@@ -481,244 +468,83 @@ SUBROUTINE SETUP_TRANS(KSMAX,KDGL,KDLON,KLOEN,LDSPLIT,PSTRET,&
+  FG%ZAS0 = FG%ZAS(:,:,IMLOC0(1))
+ENDIF
 
-ALLOCATE(ZEPSNM(D%NUMP,0:R%NTMAX+2))
-WRITE(NOUT,*)'ZEPSNM :',SIZE(ZEPSNM)
+ALLOCATE(FG%ZEPSNM(D%NUMP,0:R%NTMAX+2))
+FG%ZEPSNM = 0._JPRBT
+CALL PREPSNM !Initialize on the host
 
-ZEPSNM = 0._JPRBT
-! on the host
-CALL PREPSNM
+WRITE(NOUT,*)'setup_trans: sizes1 NUMP=',D%NUMP
 #ifdef ACCGPU
-!$ACC ENTER DATA COPYIN(ZEPSNM)
+  WRITE(NOUT,*) 'Using OpenACC'
 #endif
-
-! TODO: I guess tose might be needed again
-! add arrays for GPNORM1
-!ALLOCATE(ZAVE(IF_FS,R%NDGL))
-!ALLOCATE(ZMINGL(IF_FS,R%NDGL))
-!ALLOCATE(ZMAXGL(IF_FS,R%NDGL))
-!ALLOCATE(ZMINGPN(IF_FS))
-!ALLOCATE(ZMAXGPN(IF_FS))
-!
-!ZAVE = 0._JPRBT
-!ZMINGL = 0._JPRBT
-!ZMAXGL = 0._JPRBT
-!ZMINGPN = 0._JPRBT
-!ZMAXGPN = 0._JPRBT
-!#ifdef ACCGPU
-!!$ACC ENTER DATA COPYIN(ZAVE,ZMINGL,ZMAXGL,ZMINGPN,ZMAXGPN)
-!#endif
-
-!set up flat copies of constant data
-R_NSMAX=R%NSMAX
-R_NTMAX=R%NTMAX
-R_NDGNH=R%NDGNH
-R_NDGL=R%NDGL
-
-
-ALLOCATE(D_NSTAGT0B(SIZE(D%NSTAGT0B)))
-ALLOCATE(D_NSTAGT1B(SIZE(D%NSTAGT1B)))
-ALLOCATE(D_NPNTGTB0(0:SIZE(D%NPNTGTB0,1)-1,SIZE(D%NPNTGTB0,2)))
-ALLOCATE(D_NPNTGTB1(SIZE(D%NPNTGTB1,1),SIZE(D%NPNTGTB1,2)))
-ALLOCATE(D_MYMS(SIZE(D%MYMS)))
-ALLOCATE(D_NPROCL(SIZE(D%NPROCL)))
-ALLOCATE(D_NASM0(0:SIZE(D%NASM0)-1))
-ALLOCATE(D_NSTAGTF(SIZE(D%NSTAGTF)))
-ALLOCATE(D_MSTABF(SIZE(D%MSTABF)))
-ALLOCATE(D_NPROCM(0:SIZE(D%NPROCM)-1))
-ALLOCATE(D_NPTRLS(SIZE(D%NPTRLS)))
-
-ALLOCATE(G_NDGLU(0:SIZE(G%NDGLU)-1))
-ALLOCATE(G_NMEN(SIZE(G%NMEN)))
-ALLOCATE(G_NLOEN(SIZE(G%NLOEN)))
-
-ALLOCATE(F_RW(SIZE(F%RW)))
-ALLOCATE(F_RLAPIN(-1:SIZE(F%RLAPIN)-2))
-ALLOCATE(F_RACTHE(SIZE(F%RACTHE)))
-
-
-DO I=0,SIZE(G%NDGLU)-1
-   G_NDGLU(I)=G%NDGLU(I)
-END DO
-
-G_NMEN_MAX=0
-DO I=1,SIZE(G%NMEN)
-   G_NMEN(I)=G%NMEN(I)
-   IF (G_NMEN(I) .GT. G_NMEN_MAX) G_NMEN_MAX=G_NMEN(I)
-END DO
-
-G_NLOEN_MAX=0
-DO I=1,SIZE(G%NLOEN)
-   G_NLOEN(I)=G%NLOEN(I)
-   IF (G_NLOEN(I) .GT. G_NLOEN_MAX) G_NLOEN_MAX=G_NLOEN(I)
-END DO
-
-DO I=1,SIZE(D%NSTAGT0B)
-   D_NSTAGT0B(I)=D%NSTAGT0B(I)
-END DO
-
-DO I=1,SIZE(D%NSTAGT1B)
-   D_NSTAGT1B(I)=D%NSTAGT1B(I)
-END DO
-
-DO I=1,SIZE(D%NPROCL)
-   D_NPROCL(I)=D%NPROCL(I)
-END DO
-
-DO I=0,SIZE(D%NASM0)-1
-   D_NASM0(I)=D%NASM0(I)
-END DO
-
-DO I=1,SIZE(D%NSTAGTF)
-   D_NSTAGTF(I)=D%NSTAGTF(I)
-END DO
-
-DO I=1,SIZE(D%MSTABF)
-   D_MSTABF(I)=D%MSTABF(I)
-END DO
-
-DO I=0,SIZE(D%NPROCM)-1
-   D_NPROCM(I)=D%NPROCM(I)
-END DO
-
-DO I=1,SIZE(D%NPTRLS)
-   D_NPTRLS(I)=D%NPTRLS(I)
-END DO
-
-DO I=1,SIZE(D%NPNTGTB0,2)
-   DO J=0,SIZE(D%NPNTGTB0,1)-1
-      D_NPNTGTB0(J,I)=D%NPNTGTB0(J,I)
-   END DO
-END DO
-
-DO I=1,SIZE(D%NPNTGTB1,2)
-   DO J=1,SIZE(D%NPNTGTB1,1)
-      D_NPNTGTB1(J,I)=D%NPNTGTB1(J,I)
-   END DO
-END DO
-
-D_OFFSETS_GEMM1 => D%OFFSETS_GEMM1
-D_OFFSETS_GEMM2 => D%OFFSETS_GEMM2
 #ifdef OMPGPU
+  WRITE(NOUT,*) 'Using OpenMP offloading'
 #endif
-#ifdef ACCGPU
-!$ACC ENTER DATA COPYIN(D_OFFSETS_GEMM1,D_OFFSETS_GEMM2)
-#endif
-
-D_NUMP=D%NUMP
-D_NDGL_FS=D%NDGL_FS
+WRITE(NOUT,'(A10,":",I9,"B")') 'FG%ZAS', SIZEOF(FG%ZAS)
```

Comment:
`SIZEOF` isn't portable so must be replaced. Would printing the `SIZE` be a good substitute for this diagnostic?

Reply:
I grepped for SIZEOF, but I guess only in the internal folder. If you don't mind, I change it to SIZE(...)*C_SIZEOF(...) because I think it is good to have bytes here, because not all prints are the same precision.

Reply:
Sounds good!